src/ophys_etl/modules/segmentation/graph_utils/feature_vector_segmentation.py

```diff
@@ -0,0 +1,630 @@
+    """
+    Convert a graph into a np.ndarray image
+
+    Parameters
+    ----------
+    graph_path: pathlib.Path
+        Path to graph pickle file
+
+    attribute: str
+        Name of the attribute used to create the image
+        (default = 'filtered_hnc_Gaussian')
+
+    Returns
+    -------
+    np.ndarray
+        An image in which the value of each pixel is the
+        sum of the edge weights connected to that node in
+        the graph.
+    """
+    graph = nx.read_gpickle(graph_path)
+    node_coords = np.array(graph.nodes).T
+    row_max = node_coords[0].max()
+    col_max = node_coords[1].max()
+    img = np.zeros((row_max+1, col_max+1), dtype=float)
+    for node in graph.nodes:
+        vals = [graph[node][i][attribute] for i in graph.neighbors(node)]
+        img[node[0], node[1]] = np.sum(vals)
+    return img
+
+
+def find_a_peak(img_masked: np.ma.core.MaskedArray,
+                mu: float,
+                sigma: float,
+                n_sigma: int = 2) -> Optional[int]:
+    """
+    Find a peak in a masked, flattened image array
+
+    Parameters
+    ----------
+    img_masked: np.ma.core.MaskedArray
+        A flattened, masked image array
+
+    mu: float
+        The value taken as the mean pixel value for
+        assessing peak validity
+
+    sigma: float
+        The value taken as the standard deviation of pixel
+        values for assessing peak validity
+
+    n_sigma: int
+        The number of standard deviations a peak must deviate
+        from mu to be considered valid (default: 2)
+
+    Returns
+    -------
+    i_max: Optional[int]
+        The indes of the peak value of img_masked if
+        img_masked[i_max] > mu + n_sigma*sigma.
+        If not, return None
+    """
+    candidate = np.argmax(img_masked)
+    if img_masked[candidate] > mu+n_sigma*sigma:
+        return candidate
+    return None
+
+
+def find_peaks(img: np.ndarray,
+               mask: Optional[np.ndarray] = None,
+               slop: int = 20,
+               n_sigma: int = 2) -> List[dict]:
+    """
+    Find the peaks in an image to be used as seeds for ROI finding
+
+    Parameters
+    ----------
+    img: np.ndarray
+
+    mask: Optional[np.ndarray]
+        Optional mask set to True in every pixel that has already been
+        identified as part of an ROI. These pixels will be ignored in
+        peak finding (default: None)
+
+    slop: int
+        The number of pixels to each side of a peak that will be masked
+        out from further peak consideration on the assumption that they
+        are part of the centrail peak's ROI (default: 20)
+
+    n_sigma: int
+        The number of sigma a peak must deviate from the median of the
+        pixel brightness distribution to be considered an ROI candidate.
+        (default: 2)
+
+    Returns
+    -------
+    seeds: List[dict]
+       A list of seeds for ROI finding. Seeds are dicts of the form
```

Comment:
`NamedTuple` for this seed dict might be nice since it is used by multiple functions

Reply:
I implemented it as a TypedDict

Reply:
that's actually what I meant to say